src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -89,14 +90,28 @@
     <Warning Condition="'$(SkipNotifyEvent)' != 'true' and '$(HelixApiAccessKey)' == '' and '$(EventHubInfoMissing)' == '' " Text="EventHubPath and EventHubAccessKeys have been deprecated, use HelixApiAccessKey instead" />
     <Error Condition="'$(SkipNotifyEvent)' != 'true' and '$(HelixApiAccessKey)' == '' and '$(EventHubInfoMissing)' == 'true' " Text="HelixApiAccessKey must be set to start Helix jobs" />
 
-    <!-- gather the test archives for upload -->
-    <!-- This will include every zip in the test archive, to allow CoreCLR tests for perf which don't have a TestTFM directory -->
+    <!-- gather the test archives for upload and filter based on FilterToTestTFM -->
     <ItemGroup>
-      <ForUpload Include="$(TestArchivesRoot)**/*.zip" />
-      <ForUpload Include="$(AnyOSTestArchivesRoot)**/*.zip" />
+      <UnfilteredTestArchives Include="$(TestArchivesRoot)**/*.zip" >
+        <OSPlatformConfig>$(OSPlatformConfig)</OSPlatformConfig>
+      </UnfilteredTestArchives>
+      <UnfilteredTestArchives Include="$(AnyOSTestArchivesRoot)**/*.zip" >
+        <OSPlatformConfig>$(AnyOSPlatformConfig)</OSPlatformConfig>
+      </UnfilteredTestArchives>
       <!-- Only include Unix folders if supported by the current target OS -->
-      <ForUpload Condition="'$(TargetsUnix)' == 'true'" Include="$(UnixTestArchivesRoot)**/*.zip" />
+      <UnfilteredTestArchives Condition="'$(TargetsUnix)' == 'true'" Include="$(UnixTestArchivesRoot)**/*.zip" >
+        <OSPlatformConfig>$(UnixPlatformConfig)</OSPlatformConfig>
+      </UnfilteredTestArchives>
     </ItemGroup>
+    <FilterForUpload UnfilteredUploadItems="@(UnfilteredTestArchives)"
+             FilterToTestTFM="$(FilterToTestTFM)"
+             FilterToOSGroup="$(FilterToOSGroup)" 
```

Comment:
can we add like <Error ..> with the condition assert FilterToTestTFM/OSGroup are not empty. I assume this should be the case. right?